### ISSUE TITLE
Remove TaxonCurator Taxon association

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#3255]
I think a `TaxonCurator` should `belong_to` a `TaxonFramework`, not a `Taxon`, right @loarie? We have some comments to that effect, just want to get that cleaned up.

**Comment by loarie** [comment]
yes `belongs_to :taxon` can be removed, its been using `belongs_to :taxon_framework`
